libraries/browser-tracker-core/src/tracker/out_queue.ts

```diff
@@ -351,7 +352,14 @@ export function OutQueueManager(
       // Time out POST requests after connectionTimeout
       const xhrTimeout = setTimeout(function () {
         xhr.abort();
-        executingQueue = false;
+
+        if (retryFailures) {
+          LOG.warn(`Request failed, retrying.`);
+        } else {
+          LOG.error(`Request failed and retryFailures is false, will not retry.`);
+          removeEventsFromQueue(numberToSend);
+          executingQueue = false;
+        }
       }, connectionTimeout);
 
       const removeEventsFromQueue = (numberToSend: number): void => {
```

Comment:
Can't comment in that function but should we check for `retryFailures` in `shouldRetryForStatusCode` function? I am not sure if that flag is taken into account if we get an error status code from the collector.